deltachat-ffi/deltachat.h

```diff
@@ -3240,6 +3276,36 @@ int             dc_msg_is_setupmessage        (const dc_msg_t* msg);
 char*           dc_msg_get_setupcodebegin     (const dc_msg_t* msg);
 
 
+/**
+ * Check if the message is a videochat invitation.
+ *
+ * Typically, such messages are rendered differently by the UIs,
+ * if so, they should contain a button to join the videochat.
+ * The url for joining can be retrieved using dc_msg_get_videochat_url().
+ *
+ * @param msg The message object.
+ * @return Tristate, 0=message is no videochat invitation,
+ *     1=message is a videochat invitation that should be handled by a supported browser,
+ *     2=message is a videochat invitation that can be handled internally or by a supported browser.
```

Comment:
What's the difference between return values 1 and 2?

Reply:
1 is only an opaque-url, the UI cannot to anything on that than opening in a browser.

2 can be opened in a browser but also handled internally (eg. desktop ships the related javascript code, for android/ios native implementation not requiring a browser may come at some point). more concrete, that is an instance of https://github.com/cracker0dks/basicwebrtc

Reply:
How does the function tell between the two? By making some requests to the URL? Is this function also blocking?

Reply:
the function is not blocking. core will know by some well-known part of the url that this is basic-webrtc or the sender will tell that in some header, this is not settled yet. for now, _all_ urls will be treated as basic-webrtc, maybe we can also remote that bit for now.